src/main/java/com/atlassian/bitbucket/jenkins/internal/deployments/DeploymentPosterImpl.java

```diff
@@ -0,0 +1,88 @@
+import com.atlassian.bitbucket.jenkins.internal.client.BitbucketClientFactoryProvider;
+import com.atlassian.bitbucket.jenkins.internal.client.exception.AuthorizationException;
+import com.atlassian.bitbucket.jenkins.internal.client.exception.BitbucketClientException;
+import com.atlassian.bitbucket.jenkins.internal.config.BitbucketPluginConfiguration;
+import com.atlassian.bitbucket.jenkins.internal.config.BitbucketServerConfiguration;
+import com.atlassian.bitbucket.jenkins.internal.credentials.BitbucketCredentials;
+import com.atlassian.bitbucket.jenkins.internal.credentials.JenkinsToBitbucketCredentials;
+import com.atlassian.bitbucket.jenkins.internal.model.deployment.BitbucketCDCapabilities;
+import com.atlassian.bitbucket.jenkins.internal.model.deployment.BitbucketDeployment;
+import com.cloudbees.plugins.credentials.Credentials;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.Optional;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static java.lang.String.format;
+
+@Singleton
+public class DeploymentPosterImpl implements DeploymentPoster {
+
+    protected static final Logger LOGGER = Logger.getLogger(DeploymentPosterImpl.class.getName());
+
+    private final BitbucketClientFactoryProvider bitbucketClientFactoryProvider;
+    private final JenkinsToBitbucketCredentials jenkinsToBitbucketCredentials;
+    private final BitbucketPluginConfiguration pluginConfiguration;
+
+    @Inject
+    public DeploymentPosterImpl(BitbucketClientFactoryProvider bitbucketClientFactoryProvider,
+                                JenkinsToBitbucketCredentials jenkinsToBitbucketCredentials,
+                                BitbucketPluginConfiguration pluginConfiguration) {
+        this.bitbucketClientFactoryProvider = bitbucketClientFactoryProvider;
+        this.jenkinsToBitbucketCredentials = jenkinsToBitbucketCredentials;
+        this.pluginConfiguration = pluginConfiguration;
+    }
+
+    @Override
+    public void postDeployment(String bbsServerId, String projectKey, String repositorySlug, String revisionSha,
+                               BitbucketDeployment deployment, Run<?, ?> run,
+                               TaskListener taskListener) {
+        Optional<BitbucketServerConfiguration> maybeServer = pluginConfiguration.getServerById(bbsServerId);
+        if (!maybeServer.isPresent()) {
+            taskListener.error(format("Could not send deployment notification to Bitbucket Server: Unknown serverId %s", bbsServerId));
+            return;
+        }
+
+        BitbucketServerConfiguration server = maybeServer.get();
+        Credentials globalAdminCredentials = server.getGlobalCredentialsProvider(run.getParent())
+                .getGlobalAdminCredentials()
+                .orElse(null);
+        BitbucketCredentials credentials = jenkinsToBitbucketCredentials.toBitbucketCredentials(globalAdminCredentials);
```

Comment:
What would we do if it was `ANONYMOUS_CREDENTIALS`? From what I see in other places (e.g. `BuildStatusPoster`) we just attempt the request using whatever credentials we have.

Reply:
I was thinking that we would return and not attempt the request since we know we need credentials to post deployments and just add something to Jenkins logs. (or is it useful to have this info in BbS logs too)